specviz/plugins/model_editor/items.py

```diff
@@ -22,6 +22,11 @@ def flux(self):
 
         return self.data(self.DataRole).flux
 
+    @property
+    def spectrum(self):
+        flux = self.flux  # Update the flux
+        return super().spectrum
+
```

Comment:
Is this change related to the rest of the PR?

Reply:
This is related to the 

> I made sure that the model editor redraws a plot only if the selected plot_data_item is of type ModelDataItem.

part. It ensures that when the redraw function replaces the Spectrum1D object in this data item, the data item does not return a Spectrum1D with the wrong flux values. The `flux = self.flux` line updates the flux values of the Spectrum1D by overriding its `_data` attribute (see the `flux` property for how it does this).